projects/bp-gallery/src/components/provider/GrowthBookProvider.tsx

```diff
@@ -0,0 +1,62 @@
+import {
+  GrowthBook,
+  GrowthBookProvider as _GrowthBookProvider,
+} from '@growthbook/growthbook-react';
+import { PropsWithChildren, useEffect } from 'react';
+
+const growthbookApiHost = import.meta.env.VITE_REACT_APP_GROWTHBOOK_APIHOST;
+const growthbookClientKey = import.meta.env.VITE_REACT_APP_GROWTHBOOK_CLIENTKEY;
+
+const growthbook =
+  growthbookApiHost && growthbookClientKey
+    ? new GrowthBook({
+        apiHost: growthbookApiHost,
+        clientKey: growthbookClientKey,
+        enableDevMode: import.meta.env.MODE === 'development',
+        trackingCallback: (experiment, result) => {
+          const w: any = window;
+          const _paq: Array<any> = (w._paq = w._paq || []);
+          _paq.push([
+            'trackEvent',
+            'ExperimentViewed',
+            experiment.key,
+            'v' + String(result.variationId),
+          ]);
+        },
+        onFeatureUsage: (featureKey, result) => {
+          const w: any = window;
+          const _paq: Array<any> = (w._paq = w._paq || []);
+          _paq.push(['trackEvent', 'ExperimentViewed', featureKey, 'v' + String(result)]);
+        },
+      })
+    : undefined;
+
+export const GrowthBookProvider = ({ children }: PropsWithChildren<{}>) => {
+  useEffect(() => {
+    growthbook?.loadFeatures({ autoRefresh: true });
+  }, []);
+  useEffect(() => {
+    // Set user attributes for targeting (from cookie, auth system, etc.)
+    let visitor_id: string;
+    const w: any = window;
+    const _paq: Array<any> = (w._paq = w._paq || []);
+    _paq.push([
+      function (this: any) {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+        visitor_id = this.getVisitorId();
+        growthbook?.setAttributes({ ...growthbook.getAttributes(), id: visitor_id });
+      },
+    ]);
+
+    const refresh = (): NodeJS.Timeout | undefined => {
+      if (!growthbook) return;
+      growthbook.refreshFeatures();
+      return setTimeout(refresh, 5000);
+    };
+
+    const r = refresh();
+    return () => clearTimeout(r);
```

Comment:
This will probably not work after the second time `refresh` was called, since `r` will always refer to the initial call. You should probably declare `r` as `let` and update it inside `refresh`. While you're at it, maybe rename `r` to something more readable :D. All in all, something like this:
```ts
let timeout = 0;
const refresh = () => {
  if (!growthbook) return;
  growthbook.refreshFeatures();
  timeout = window.setTimeout(refresh, 5000);
};

refresh();
return () => window.clearTimeout(timeout);
```